protocol/x/listing/keeper/msg_server_upgrade_isolated_perpetual_to_cross_test.go

```diff
@@ -0,0 +1,60 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	types "github.com/dydxprotocol/v4-chain/protocol/x/listing/types"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// validAuthority is a valid bech32 address.
+	validAuthority = constants.AliceAccAddress.String()
+)
+
+func TestMsgUpgradeIsolatedPerpetualToCross_ValidateBasic(t *testing.T) {
+	tests := map[string]struct {
+		msg         types.MsgUpgradeIsolatedPerpetualToCross
+		expectedErr string
+	}{
+		"Success": {
+			msg: types.MsgUpgradeIsolatedPerpetualToCross{
+				Authority:   validAuthority,
+				PerpetualId: 1,
+			},
+		},
+		"Failure: Empty authority": {
+			msg: types.MsgUpgradeIsolatedPerpetualToCross{
+				Authority:   "",
+				PerpetualId: 1,
+			},
+			expectedErr: "Authority is invalid",
+		},
+		"Failure: Invaid authority": {
+			msg: types.MsgUpgradeIsolatedPerpetualToCross{
+				Authority:   "invalid",
+				PerpetualId: 1,
+			},
+			expectedErr: "Authority is invalid",
+		},
+		"Failure: Invaid perpetual ID": {
+			msg: types.MsgUpgradeIsolatedPerpetualToCross{
+				Authority:   validAuthority,
+				PerpetualId: 0,
+			},
+			expectedErr: "Invalid perpetual ID",
+		},
+	}
```

Comment:
Can you also add a test for a perpetual id that doesn't exist

Reply:
These are supposed to be basic validation tests on the msg fields IIUC? The next PR has actual functionality tests

Reply:
Chatted offline, `ValidateBasic` convention for Cosmos has been deprecated, msg should be validated within the actual msg execution. So both msg validation and execution will be tested here in this PR